src/core/createEvent.js

```diff
@@ -52,6 +52,17 @@ export default () => {
     set lastChanceCallback(value) {
       lastChanceCallback = value;
     },
+    validate() {
+      const warnings = [];
+      if (event.isEmpty()) {
+        warnings.push("No event xdm or event data specified.");
+      }
+      event.toJSON();
```

Comment:
I don't think we should be calling `toJSON` because it is currently being calling at a very specific point in the lifecyle so we can merge the event command's data with other data provided by the onBeforeSend callback and auto collected data.. 

@Aaronius @jonsnyder

Reply:
Yes, toJSON is doing a lot of things it probably shouldn't do given the function name by altering the event internals. However, I'll change the code to not rely on toJSON.